int32 agent_num
float32[] ck_array